From bf016bb65106397f845db70009bedac6eff41f1b Mon Sep 17 00:00:00 2001
From: Tim Pope <code@tpope.net>
Date: Tue, 13 May 2008 15:22:48 -0400
Subject: [PATCH] Fix tests broken by mocha absence

Two ActiveRecord tests depended on mocha but were not marked as such.
Place them in a use_mocha block so the test suite passes.
---
 activerecord/test/cases/fixtures_test.rb |   20 +++++++++++---------
 1 files changed, 11 insertions(+), 9 deletions(-)

diff --git a/activerecord/test/cases/fixtures_test.rb b/activerecord/test/cases/fixtures_test.rb
index 2787b9a..aca7cfb 100755
--- a/activerecord/test/cases/fixtures_test.rb
+++ b/activerecord/test/cases/fixtures_test.rb
@@ -610,15 +610,17 @@ def test_foo
 end
 
 class FixtureLoadingTest < ActiveRecord::TestCase
-  def test_logs_message_for_failed_dependency_load
-    Test::Unit::TestCase.expects(:require_dependency).with(:does_not_exist).raises(LoadError)
-    ActiveRecord::Base.logger.expects(:warn)
-    Test::Unit::TestCase.try_to_load_dependency(:does_not_exist)
-  end
+  uses_mocha 'reloading_fixtures_through_accessor_methods' do
+    def test_logs_message_for_failed_dependency_load
+      Test::Unit::TestCase.expects(:require_dependency).with(:does_not_exist).raises(LoadError)
+      ActiveRecord::Base.logger.expects(:warn)
+      Test::Unit::TestCase.try_to_load_dependency(:does_not_exist)
+    end
 
-  def test_does_not_logs_message_for_successful_dependency_load
-    Test::Unit::TestCase.expects(:require_dependency).with(:works_out_fine)
-    ActiveRecord::Base.logger.expects(:warn).never
-    Test::Unit::TestCase.try_to_load_dependency(:works_out_fine)
+    def test_does_not_logs_message_for_successful_dependency_load
+      Test::Unit::TestCase.expects(:require_dependency).with(:works_out_fine)
+      ActiveRecord::Base.logger.expects(:warn).never
+      Test::Unit::TestCase.try_to_load_dependency(:works_out_fine)
+    end
   end
 end
-- 
1.5.5.1

